From 787ee1a937f1d52e009bd33503f520d1410093e0 Mon Sep 17 00:00:00 2001
From: Vitaly Kleban <vk@everynet.com>
Date: Thu, 15 Dec 2022 21:17:34 +0200
Subject: [PATCH 2/3] Vector_{erase, remove} wrong pointer casting

---
 vec.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vec.h b/vec.h
index 4c2d586..ce9e43f 100644
--- a/vec.h
+++ b/vec.h
@@ -50,9 +50,9 @@ typedef char* vec_char;
 
 // vec is a vector (aka type*)
 #define vector_erase(vec, pos, len)                                            \
-	(_vector_erase((vector*)vec, sizeof(*vec), pos, len))
+	(_vector_erase((vector*)&vec, sizeof(*vec), pos, len))
 #define vector_remove(vec, pos)                                                \
-	(_vector_remove((vector*)vec, sizeof(*vec), pos))
+	(_vector_remove((vector*)&vec, sizeof(*vec), pos))
 
 #define vector_copy(vec) (_vector_copy((vector*)vec, sizeof(*vec)))
 
-- 
2.41.0

